### DISCUSSION TITLE
How to "oh-my-posh upgrade" where current binary is in directory writable only by root?

### DISCUSSION THREAD

**Discussion by GethDeeo** [JanDeDobbeleer/oh-my-posh#4991] [Troubleshoot] [answered]
Current binary is in `/usr/local/bin`, which is writable only by root.

```
$ sudo oh-my-posh upgrade
[sudo] password for geth:

    ⚠️  we don't have permissions to upgrade oh-my-posh, please use elevated permissions to upgrade

```
Beginner here... I cannot elevate it further :)

**Comment by JanDeDobbeleer** [comment]
@GethDeeo `sudo oh-my-posh upgrade`

**Reply by PGScully** [reply]
I got the same error both with and without sudo.

**Reply by GethDeeo** [reply]
@JanDeDobbeleer That's what I've tried as in the discussion's body.

**Reply by JanDeDobbeleer** [reply, marked as answer]
@PGScully it doesn't work on Linux. Apparently a program can't update itself as the file handle will be blocked from writing.

**Reply by diogo-rossi** [reply]
@PGScully
In this situation, I usually  do:
```
wget https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/posh-linux-amd64 -O /path/to/writable/folder/oh-my-posh
```
And `/path/to/writable/folder/` has to be a writable directory and added to your `PATH` variable manually.
